### PR TITLE
tests: shellchecks, part 1

### PR DESCRIPTION
**Pull request by bboozzoo**
Part 1 of shellchecks.

### CODE REVIEW

File: tests/completion/indirect/task.yaml [hunk 1]
```diff
@@ -12,16 +12,17 @@ prepare: |
...
-      cd ../../lib/snaps/test-snapd-complexion
+      cd ../../lib/snaps/test-snapd-complexion || exit 1
       mv test-snapd-complexion.bash-completer.orig test-snapd-complexion.bash-completer
       snap remove test-snapd-complexion
   )
 
 execute: |
   d="$PWD"
+  #shellcheck disable=SC1090
   source "${SPREAD_PATH}/${SPREAD_SUITE}/${SPREAD_VARIANT}.vars"
```
Comment (zyga) [inline tests/completion/indirect/task.yaml]: Can we just use . instead of source here?

Comment (zyga) [reply tests/completion/indirect/task.yaml]: After reading https://github.com/koalaman/shellcheck/wiki/SC1090 I think this is fine.

File: tests/completion/simple/task.yaml [hunk 2]
```diff
@@ -2,6 +2,7 @@ summary: simple completion
 
 execute: |
   d="$PWD"
+  #shellcheck disable=SC1090
   source "${SPREAD_PATH}/${SPREAD_SUITE}/${SPREAD_VARIANT}.vars"
```
Comment (zyga) [inline tests/completion/simple/task.yaml]: For reference: https://github.com/koalaman/shellcheck/wiki/SC1090

File: tests/main/auto-refresh-private/task.yaml [hunk 3]
```diff
@@ -52,18 +54,22 @@ execute: |
     done
 
     echo "Check they were both refreshed"
+    #shellcheck disable=SC1117
```
Comment (zyga) [inline tests/main/auto-refresh-private/task.yaml]: Hmm, isn't this supposed to be `\\` Alternatively can we use `'test-snapd-public +2\.0'`?

File: tests/main/auto-refresh-private/task.yaml [hunk 4]
```diff
@@ -52,18 +54,22 @@ execute: |
...
     snap list|MATCH "test-snapd-public +2\.0"
+    #shellcheck disable=SC1117
```
Comment (zyga) [inline tests/main/auto-refresh-private/task.yaml]: Same as above

File: tests/main/classic-confinement-not-supported/task.yaml [hunk 5]
```diff
@@ -28,11 +30,13 @@ execute: |
     if [[ "$SPREAD_SYSTEM" = fedora-* ]]; then
         EXPECTED_TEXT="classic confinement requires snaps under /snap or symlink from /snap to /var/lib/snapd/snap"
     fi
+    #shellcheck disable=SC2015
```
Comment (zyga) [inline tests/main/classic-confinement-not-supported/task.yaml]: For reference: https://github.com/koalaman/shellcheck/wiki/SC2015

Comment (zyga) [reply tests/main/classic-confinement-not-supported/task.yaml]: Is it telling us that the || true is not needed in practice?
Unless `$(...)` inherits `set -e` that is, but does it?

File: tests/main/abort/task.yaml [hunk 6]
```diff
@@ -17,11 +18,13 @@ execute: |
     echo "Abort with valid id - error"
     subdirPath="$SNAP_MOUNT_DIR/$SNAP_NAME/current/foo"
     mkdir -p "$subdirPath"
+    #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB/snaps.sh"
     if install_local "$SNAP_NAME"; then
         echo "install should fail when the target directory exists"
         exit 1
     fi
+    #shellcheck disable=SC1117
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply tests/main/abort/task.yaml]: Ok, let me review the places. Where possible I'd prefer to use '...\<specialchar>..'. Shell quoting rules, annoying as always.

File: tests/main/classic-confinement-not-supported/task.yaml (hunk 5 above)
Comment (bboozzoo) [reply tests/main/classic-confinement-not-supported/task.yaml]: I'll drop the whole `.. && exit 1 || true` as it makes little sense here.

**Comment by zyga** [comment]
I saw an error like this (it was unclear which test caused it):
```
/dev/stdin: line 1: syntax error near unexpected token `newline'
/dev/stdin: line 1: `<!DOCTYPE html>'

Crushing failure and despair.
```

I re-started as it looks as some 404 page or other web issue

**Pull request merged**